libs/application/templates/new-primary-school/src/lib/newPrimarySchoolUtils.ts

```diff
@@ -499,10 +437,18 @@ export const getOptionsListByType = async (
 
   return (
     data?.friggOptions?.flatMap(({ options }) =>
-      options.flatMap(({ value, id }) => {
-        const content = value.find(({ language }) => language === 'is')?.content
-        return { value: id ?? '', label: content ?? '' }
+      options.flatMap(({ value, key }) => {
+        const content = value.find(({ language }) => language === lang)?.content
+        return { value: key ?? '', label: content ?? '' }
       }),
     ) ?? []
   )
 }
+
+export const getSelectOptionLabel = (options: SelectOption[], key?: string) => {
```

Comment:
Wouldn't it make more sense if it's 'selected' instead of 'select'
```suggestion
export const getSelectedOptionLabel = (options: SelectOption[], key?: string) => {
```

Reply:
Fixed